nat-lab/tests/utils/batching.py

```diff
@@ -0,0 +1,162 @@
+# There's a dillemma in how to best capture the traffic. Capturing on the node itself
+# provides insights into the packets themselves coming to and from the tunnel device,
+# however capturing locally doesn't mean the packet actually left the machine.
+# An alternative is to capture on gateways which guarantees the packet arrival
+# however there's no way to easily pry open the packets and observe their insides.
+import asyncio
+import math
+import os
+import pytest
+import subprocess
+import typing
+from scapy.all import PcapReader  # type: ignore
+from scapy.layers.inet import IP, ICMP  # type: ignore
+from typing import Any, Callable, List, Tuple
+
+
+class Histogram:
+    def __init__(self) -> None:
+        self._values: typing.List[float] = []
+
+    def add_value(self, value: float) -> None:
+        self._values.append(value)
+
+    def get(self, bin_count: int, bin_width: int) -> List[int]:
```

Comment:
Nit: I think it's more common to talk about buckets than about bins when it comes to histograms

Reply:
Good point to be honest. I've renamed all `bin` occurences into `bucket`